libr/asm/p/asm_x86_nz.c

```diff
@@ -4913,6 +4910,15 @@ static int parseOperand(RAsm *a, const char *str, Operand *op, bool isrepop) {
 			str = ++p;
 		}
 		op->immediate = getnum (a, str);
+		if (op->immediate == -1) {
+			op->is_good_flag = false;
+			char *num_str = strdup (str);
```

Comment:
use r_str_trim_dup() and save one line

Reply:
ok 185eec4a44cd94f7a9175a7d41fcdca1020898ad